### PR TITLE
Feat calypso op

### PR DESCRIPTION
**Pull request by wangzyphysics**

### CODE REVIEW

**Comment by wangzyphysics** [comment]
4 calypso related ops have been added:
- `prep_caly_input.py`: prep the input file of calypso according to the given List[List[dict]].
- `collect_run_caly.py`: make calypso run to generate structures depend on `input.dat` in the beginning, and other steps run depend on `input.dat`, `results/`, `step`, `POSCAR_*`, `OUTCAR_*` and `CONTCAR_*`.
- `prep_run_dp_optim.py`: make structure optimization with dp.
- `run_caly_model_devi.py`: make model deviation calculation of selected trajectory frames.

And 4 ops compose 2 superpos：
The superop `caly_evo_step.py` consists of the second and third op.
The super op `prep_run_calypso.py` consists of the first op, the super `caly_evo_step.py` and the forth op.

File: dpgen2/op/prep_caly_input.py [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+    A calypso input file will be generated according to the given parameters
+    (defined by `ip["caly_inputs"]`). The artifact will be return
+    (ip[`input_files`]). The name of directory is `ip["task_names"]`.
+    """
+
+    @classmethod
+    def get_input_sign(cls):
+        return OPIOSign(
+            {
+                "caly_task_grp": BigParameter(Path),  # calypso input params
```
Comment (zjgemi) [inline dpgen2/op/prep_caly_input.py:309]: Make sure the type is Path?

File: dpgen2/entrypoint/submit.py [hunk 2]
```diff
@@ -199,6 +226,61 @@ def make_naive_exploration_scheduler(
...
+
+    for job_ in model_devi_jobs:
+        if not isinstance(job_, list):
+            job = [job_]
+        else:
+            job = job_
+        # stage
+        stage = ExplorationStage()
+        for jj in job:
+            # jconf = normalize_task_group_config(jj)
```
Comment (wanghan-iapcm) [inline dpgen2/entrypoint/submit.py:264]: why skip normalizing the configuration?

File: dpgen2/superop/block.py [hunk 3]
```diff
@@ -81,34 +85,44 @@ class ConcurrentLearningBlock(Steps):
...
             "iter_data": InputArtifact(),
         }
+        # if exploration_style == "lmp":
+        #     self._input_parameters.update(
+        #         {
+        #             "lmp_task_grp": InputParameter(),
+        #         }
+        #     )
+        # elif exploration_style == "calypso":
+        #     pass
```
Comment (wanghan-iapcm) [inline dpgen2/superop/block.py:125]: remove?

File: dpgen2/superop/block.py [hunk 4]
```diff
@@ -226,22 +240,41 @@ def _block_cl(
...
-        },
-        artifacts={
-            "models": prep_run_dp_train.outputs.artifacts["models"],
-        },
-        key="--".join(
-            ["%s" % block_steps.inputs.parameters["block_id"], "prep-run-lmp"]
-        ),
-    )
-    block_steps.add(prep_run_lmp)
+    if block_steps.explore_style == "lmp":
```
Comment (wanghan-iapcm) [inline dpgen2/superop/block.py:243]: I guess the two branches may be merged.  we may change the original lmp_config->expl_config and lmp_task_grp -> expl_task_grp. 
you may need to add a dummy type_map  parameter to prep_run_lmp

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]
- the exploration/task module needs rearrangement, i will take care in a separate PR. 
- please provide an example for the calypso exploration.

File: dpgen2/superop/prep_run_lmp.py [hunk 5]
```diff
@@ -58,8 +58,9 @@ def __init__(
     ):
         self._input_parameters = {
             "block_id": InputParameter(type=str, value=""),
-            "lmp_config": InputParameter(),
-            "lmp_task_grp": InputParameter(),
+            "explore_config": InputParameter(),
+            "expl_task_grp": InputParameter(),
+            "type_map": InputParameter(value=""),
```
Comment (wanghan-iapcm) [inline dpgen2/superop/prep_run_lmp.py:63]: default value being `[]`?

**Review by wanghan-iapcm — COMMENTED** [review]
please fix the pyright check.

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**